internal/internal_workflow.go

```diff
@@ -862,16 +859,21 @@ func (s *coroutineState) unblocked() {
 	s.keptBlocked = false
 }
 
-func (s *coroutineState) call() {
+func (s *coroutineState) call(timeout time.Duration) {
 	s.unblock <- func(status string, stackDepth int) bool {
 		return false // unblock
 	}
 
-	deadlockDetectorTimeout := time.Second
-	if debugMode {
-		deadlockDetectorTimeout = math.MaxInt64
+	// Defaults are populated in the worker options during worker startup, but test environment
+	// may have no default value for the deadlock detection timeout, so we also need to set it here for
+	// backwards compatibility.
```

Comment:
This looks like a hack to me. Why to not initialize it in the test environment instead?

Reply:
Users might be overriding worker options after creation of the test environment and it would break as deadlock detection timeout would default to 0 in that case.